libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1942,3 +1942,19 @@ bool AP_GPS_UBLOX::supports_F9_config(void) const
 {
     return _hardware_generation == UBLOX_F9 && _hardware_generation != UBLOX_UNKNOWN_HARDWARE_GENERATION;
 }
+
+void AP_GPS_UBLOX::inject_data(const uint8_t *data, uint16_t len)
```

Comment:
please add a allow_inject_unconfigured() virtual method, defaulting to return true; so that backends that want to disallow injection while not configured can do so with an override of that method